File: src/com/timepath/quakec/QCCompletionContributor.java
package com.timepath.quakec;

import com.intellij.codeInsight.completion.*;
import com.intellij.codeInsight.lookup.LookupElementBuilder;
import com.intellij.patterns.PlatformPatterns;
import com.intellij.util.ProcessingContext;
import com.timepath.quakec.psi.QCTypes;
import org.jetbrains.annotations.NotNull;

/**
 * @author TimePath
 */
public class QCCompletionContributor extends CompletionContributor {
    public QCCompletionContributor() {
        extend(CompletionType.BASIC,
                PlatformPatterns.psiElement(QCTypes.ID).withLanguage(QCLanguage.INSTANCE),
                new CompletionProvider<CompletionParameters>() {
                    public void addCompletions(@NotNull CompletionParameters parameters,
                                               ProcessingContext context,
                                               @NotNull CompletionResultSet resultSet) {
                        resultSet.addElement(LookupElementBuilder.create("Hello"));
                    }
                }
        );
    }
}

File: src/main/kotlin/com/timepath/quakec/ide/QCCompletionContributor.kt
package com.timepath.quakec.ide

import com.intellij.codeInsight.completion.*
import com.intellij.codeInsight.lookup.LookupElementBuilder
import com.intellij.patterns.PlatformPatterns
import com.intellij.util.ProcessingContext
import com.timepath.quakec.psi.QCTypes

public class QCCompletionContributor : CompletionContributor() {
    init {
        extend(CompletionType.BASIC, PlatformPatterns.psiElement(QCTypes.TOKEN_IDENTIFIER).withLanguage(QCLanguage), object : CompletionProvider<CompletionParameters>() {
            override fun addCompletions(parameters: CompletionParameters, context: ProcessingContext, resultSet: CompletionResultSet) {
                resultSet.addElement(LookupElementBuilder.create("Hello"))
            }
        })
    }
}
